src/main/java/com/exasol/repository/MavenPomParser.java

```diff
@@ -0,0 +1,163 @@
+        if (build == null) {
+            return Optional.empty();
+        } else {
+            return parsePlugins(build);
+        }
+    }
+
+    private Optional<String> parsePlugins(final Element build) {
+        final NodeList plugins = build.getElementsByTagName("plugins").item(0).getChildNodes();
+        for (int i = 0; i < plugins.getLength(); i++) {
+            final Node next = plugins.item(i);
+            if (next.getNodeType() == Node.ELEMENT_NODE) {
+                final Element plugin = (Element) next;
+                final Node artifactIdNode = plugin.getElementsByTagName("artifactId").item(0);
+                final String artifactId = artifactIdNode.getTextContent().strip();
+                if (artifactId.equals("maven-assembly-plugin")) {
+                    return parseMavenAssemblyPlugin(plugin);
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<String> parseMavenAssemblyPlugin(final Element plugin) {
+        final Element configurations = (Element) plugin.getElementsByTagName("configuration").item(0);
+        if (configurations == null) {
+            return Optional.empty();
+        } else {
+            return getParseConfigurations(configurations);
+        }
+    }
+
+    private Optional<String> getParseConfigurations(final Element configurations) {
+        final Node finalNameNode = configurations.getElementsByTagName("finalName").item(0);
+        if ((finalNameNode == null) || (finalNameNode.getTextContent() == null)
+                || (finalNameNode.getTextContent().isEmpty())) {
+            return Optional.empty();
+        } else {
+            return parseFinalName(finalNameNode);
+        }
+    }
+
+    private Optional<String> parseFinalName(final Node finalNameNode) {
+        String finalName = finalNameNode.getTextContent().strip();
+        while (finalName.contains("${")) {
+            finalName = replaceVariable(finalName);
+        }
+        return Optional.of(finalName);
+    }
+
+    private String replaceVariable(final String finalName) {
+        final int startIndex = finalName.indexOf("${") + 2;
+        final int endIndex = finalName.indexOf('}');
+        final String tag = finalName.substring(startIndex, endIndex);
+        final String replacement = findReplacement(tag);
+        return finalName.replace("${" + tag + "}", replacement);
+    }
+
+    private String findReplacement(final String tag) {
+        if (tag.equals("version")) {
+            return parseChildElement("version");
+        } else {
+            final Element properties = (Element) getMandatoryChildNode("properties");
+            final Node tagNode = properties.getElementsByTagName(tag).item(0);
+            if (tagNode == null) {
+                throw throwParsingException(tag);
+            } else {
+                return tagNode.getTextContent().strip();
```

Comment:
We don't want a version with spaces. It shouldn't happen, so the strip here is just for safety

Reply:
I see. Still kind of dangerous since if the tag contains spaces, and you inject that value elsewhere in the POM you might get unexpected results. On the other hand that is not RR's problem. So keep it like it is.